src/main/java/service/implementation/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package service.implementation;
+
+import db.FruitStorage;
+import exception.InvalidDataFormatException;
+import java.util.List;
+import java.util.Map;
+import model.Fruit;
+import model.FruitTransactionDto;
+import model.Operation;
+import service.FruitShopService;
+import strategy.OperationStrategy;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private final Map<Operation, OperationStrategy> operationStrategyMap;
+
+    public FruitShopServiceImpl(Map<Operation, OperationStrategy> operationStrategyMap) {
+        this.operationStrategyMap = operationStrategyMap;
+    }
+
+    @Override
+    public void applyOperationsOnFruitsDto(List<FruitTransactionDto> fruitTransactionDtoList) {
+        for (FruitTransactionDto transactionDto : fruitTransactionDtoList) {
+            try {
+                Operation operation = transactionDto.getOperation();
+                operationStrategyMap.get(operation).apply(transactionDto);
+            } catch (RuntimeException e) {
+                throw new InvalidDataFormatException("Your data has invalid format,"
+                        + " something is missing" + "\n" + e);
```

Comment:
you can declare a contstant field with exception message

Reply:
Done - now it's clearer)